pkg/ring/kv/client.go

```diff
@@ -97,19 +99,19 @@ type Client interface {
 
 // NewClient creates a new Client (consul, etcd or inmemory) based on the config,
 // encodes and decodes data for storage using the codec.
-func NewClient(cfg Config, codec codec.Codec) (Client, error) {
+func NewClient(name string, cfg Config, codec codec.Codec, reg prometheus.Registerer) (Client, error) {
```

Comment:
I think passing `name` to components that is used just for metrics registration is an anti-pattern. Caller of this call can easily wrap `reg` into `prometheus.WrapRegistererWith(prometheus.Labels{"name": "some name here"}, reg)` if it wishes to do so. I would prefer to use this pattern instead. KV store client doesn't need a name for anything else, and it also doesn't need it if registrerer is `nil`.